src/main/java/Duke.java

```diff
@@ -1,10 +1,264 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.File;
+import java.io.FileNotFoundException;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+
+    protected static Scanner sc;
+    protected static ArrayList<Task> taskList;
+    protected static boolean byeFlag = false;
+    protected static final String filePath = "data/duke.txt";
+
+    public static void main(String[] args) throws IOException {
+        taskList = new ArrayList<>();
+        sc = new Scanner(System.in);
+
+        try {
+            readFileContents(filePath);
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found!!!");
+        } catch (DukeException e) {
+            System.out.println("OOPS!!! Invalid time and date format read.");
+        }
+
+        while (!byeFlag) {
+            try {
+                duke();
+            } catch (DukeException ex) {
+                System.err.println(ex);
+            }
+        }
+
+        try {
+            writeToFile(filePath);
```

Comment:
Base on my understanding of your code (I did not have the chance to run it so my understanding might not be right), the writeToFile() method only occurs when one exits the program. However, base on my understanding of the project requirement, the writeToFile() should be done whenever there is a change to the list i.e. when there is an add or delete command. Perhaps, we have different understanding of the project requirement?